### PR TITLE
feat(site-header-core): Adding component, FRONT-1798

### PR DESCRIPTION
**Pull request by planctus**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] `package.json` is up-to-date and `@ecl/[system]-base` is part of the dependencies
- [ ] I have checked the dependencies
- [ ] I have given the fractal status “ready” to my component
- [ ] I have declared `@define mycomponent` in the SCSS file
- [ ] I have specified `margin: 0;` on the CSS component
- [ ] I have provided tests
- [ ] I follow the naming guidelines
- [ ] the component supports composition
- [ ] there are no hardcoded strings (all content come from the context)
- [ ] I have filled the README.md file (at least a few lines)
- [ ] My component is listed in the root README
- [ ] My PR has the right label(s)
- [ ] The redirects in `src/website/src/routes/Redirects.jsx` are up-to-date

### CODE REVIEW

**Review by emeryro — CHANGES_REQUESTED** [review]
Component is fine, just the tests to be updated

File: src/implementations/vanilla/components/site-header-core/site-header-core-print.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@use "sass:map";
+@use '@ecl/theme-dev/maps/spacing';
+
+.ecl-site-header-core__logo-image {
+  height: 1.9cm;
+  margin-bottom: map.get(spacing.$spacing-print, '4xl');
+  margin-top: map.get(spacing.$spacing-print, '4xl');
+  padding-bottom: map.get(spacing.$spacing-print, '4xl');
+  padding-top: map.get(spacing.$spacing-print, '4xl');
```
Comment (kalinchernev) [inline src/implementations/vanilla/components/site-header-core/site-header-core-print.scss]: In v2, these are

```scss
    padding-bottom: $ecl-spacing-print-xl;
    padding-top: $ecl-spacing-print-xl;
```

File: src/implementations/vanilla/components/site-header-core/site-header-core.scss [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  transform: translateX(-50%) translateX(-0.01rem) translateY(-50%)
+    translateY(-0.05rem);
+}
+
+.ecl-site-header-core__search--active {
+  display: flex;
+}
+
+/* stylelint-disable-next-line order/order */
+@include breakpoints.up(s) {
```
Comment (kalinchernev) [inline src/implementations/vanilla/components/site-header-core/site-header-core.scss]: The rest of the implementations have strings as parameters. 
```
@include breakpoints.up('s')
```

File: src/implementations/vanilla/components/site-header-core/site-header-core.scss [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+  .ecl-site-header-core__logo-image {
+    height: $_logo-height-sm;
+    max-height: 100%;
+    max-width: 100%;
+    width: $_logo-width-sm;
+  }
+}
+
+@include breakpoints.up(l) {
```
Comment (kalinchernev) [inline src/implementations/vanilla/components/site-header-core/site-header-core.scss]: Same comment for string input parameters here.

**Review by kalinchernev — CHANGES_REQUESTED** [review]
Minor comments

**Review by kalinchernev — APPROVED** [review]

**Pull request merged**